lebedigital/raw_data_processing/metadata_extraction/mixture_metadata_extraction.py

```diff
@@ -0,0 +1,144 @@
+        
+    """
+    
+    excelsheet = os.path.basename(locationOfRawData)
+    print("\n Working on: "+ excelsheet)
+    excelfile = pd.read_excel(locationOfRawData, sheet_name= None) 
+    listofkeys = [i for i in excelfile.keys() if "Rezeptur" in i] 
+    print("Relevant sheets in this file: " + str(listofkeys))
+
+
+    for sheet in listofkeys:
+
+        name = os.path.basename(excelsheet).split(".xl")[0] + " ___ " + sheet
+        exceltodf = excelfile[sheet]
+        
+        # extract header, translate and set proper index & column headers
+        exceltodf.iat[17,4] = "Dichte"
+        if english == True:
+            exceltodf = translate(exceltodf,"translation.xlsx")
+            specimeninfo = {
+                    "Date" : str(exceltodf.columns[9])[:10], 
+                    "Editor" : exceltodf.iloc[0,10], 
+                    "Requester" : exceltodf.iloc[2,3],
+                    "Project number": exceltodf.iloc[3,3],
+                    "Specimen" : exceltodf.iloc[4,3], 
+                    "Betonsorte u -festigkeitsklasse" : exceltodf.iloc[6,4],
+                    "Wasserzementwert" : exceltodf.iloc[7,2],
+                    "Konsistenz" : exceltodf.iloc[7,7],
+                    "Sieblinie n. DIN 1045" : exceltodf.iloc[8,2],
+                    "Körnungsziffer" : exceltodf.iloc[8,8],
+                    "Vol Leim/Zuschlag" : exceltodf.iloc[10,10]
+                    }
+            word = "Supplementary cementious materials"
+        else:
+            specimeninfo = {
+                    "Datum" : str(exceltodf.columns[9])[:10], 
+                    "Bearbeiter" : exceltodf.iloc[0,10], 
+                    "Antragsteller" : exceltodf.iloc[2,3],
+                    "Antrags-/Projektnummer": exceltodf.iloc[3,3],
+                    "Bezeichnung der Proben" : exceltodf.iloc[4,3], 
+                    "Betonsorte u -festigkeitsklasse" : exceltodf.iloc[6,4],
+                    "Wasserzementwert" : exceltodf.iloc[7,2],
+                    "Konsistenz" : exceltodf.iloc[7,7],
+                    "Sieblinie n. DIN 1045" : exceltodf.iloc[8,2],
+                    "Körnungsziffer" : exceltodf.iloc[8,8],
+                    "Vol Leim/Zuschlag" : exceltodf.iloc[10,10]
+                    }
+            word = "Zusatzstoff"
+        specimeninfo = pd.Series(specimeninfo)
+        specimeninfo = specimeninfo.fillna("---") # fill nans
+        exceltodf.rename(columns=exceltodf.iloc[17], inplace = True) # set column header
+        exceltodf["Stoffart"] = exceltodf["Stoffart"].str.strip()  # remove whitespace
+        exceltodf["Sonstiges/Bemerkungen"] = exceltodf["Sonstiges/Bemerkungen"].str.strip()  # remove whitespace
+        
+        # Some files have multiple entries of Zusatzstoffe, which are not further labeled, so this adds 
+        # the type of Zusatzstoff
+        howmanyzusatzstoffe = [True if i == word else False for i in exceltodf["Stoffart"] ]
```

Comment:
See above, my suggestion is to use a dict such as 
`exceltopdf[translate[language]['type']]` with language being set to german if the read in file is a German raw data file.